test/integration/viewer_spec.mjs

```diff
@@ -454,4 +454,346 @@ describe("PDF viewer", () => {
       );
     });
   });
+
+  describe("Detail view on zoom", () => {
+    function setupPages(zoom, maxCanvasPixels, width, height) {
+      const { resolve, reject, promise } = Promise.withResolvers();
+
+      let pages;
+
+      beforeAll(async () => {
+        const pagesP = loadAndWait(
+          "colors.pdf",
+          null,
+          zoom,
+          {
+            // TODO: Why is this needed?
+            earlySetup: () => {
+              window.devicePixelRatio = 2;
+            },
```

Comment:
What sort of errors happen without this TODO?

Note that the `devicePixelRatio` value will affect the canvas-sizes in the viewer, and thus how "quickly" we reach e.g. the `maxCanvasPixels` value.
Is it possible that the initial/expected values in these tests "simply" need to take the `devicePixelRatio` into account instead?

Reply:
If I don't explicitly set the same `devicePixelRatio` in `window.devicePixelRatio` and in the viewport options for puppeteer, then it seems like in Chrome the `window.devicePixelRatio` variable doesn't actually match the pixel ratio that the browser is running with.

My screen has a pixel ratio of 2, puppeteer defaults to 1, and they seem to somehow lead to chrome scrolling to the wrong amount of pixels when trying to scroll.

I have hard-coded 2 for now and tests are passing, but I need to figure out why there is a problem and maybe run them both with 1 and non-1.